examples/apps/color_finder.py

```diff
@@ -0,0 +1,710 @@
+        num_cols (int): the number of columns in the matrix, specified in ColorFinder as downsize_width
+        num_rows (int): the number of rows in the matrix, specified in ColorFinder as downsize_height
+
+    '''
+    def __init__(self, initial_value, num_cols, num_rows):
+        self.num_cols = num_cols
+        self.num_rows = num_rows
+        self._matrix = [[MatrixValueContainer(initial_value) for _ in range(self.num_rows)] for _ in range(self.num_cols)]
+
+    def at(self, i, j):
+        '''Gets the desired MatrixValueContainer object.
+
+        Args:
+            i (int): the x-coordinate in self
+            j (int): the y-coordinate in self
+
+        Returns:
+            self._matrix[i][j] (MatrixValueContainer): the MatrixValueContainer at the specified coordinates
+
+        '''
+        return self._matrix[i][j]
+
+    def surrounded(self, i, j):
+        '''Checks if a coordinate is surrounded 
+            by at least 3 coordinates of the same value.
+
+        Args:
+            i (int): the x-coordinate in self._matrix
+            j (int): the y-coordinate in self._matrix
+
+        Returns:
+            the surrounding value if the condition is True, otherwise returns None
+
+        '''
+        if i != 0 and i != self.num_cols-1 and j != 0 and j != self.num_rows-1:
+            left_value, up_value, right_value, down_value = self.get_neighboring_values(i, j)
+            if left_value == up_value and left_value == right_value:
+                return left_value
+            if left_value == up_value and left_value == down_value:
+                return left_value
+            if left_value == right_value and left_value == down_value:
+                return left_value
+            if right_value == up_value and right_value == down_value:
+                return right_value
+        return None
+
+    def get_neighboring_values(self, i, j):
+        '''Returns the values in the four surrounding MatrixValueContainers.
+        
+        Args:
+            i (int): the x-coordinate in self._matrix
+            j (int): the y-coordinate in self._matrix
+
+        Returns:
+            A four-tuple containing (left_value, up_value, right_value, and down_value)
+
+        '''
+        return (self.at(i-1, j), self.at(i, j-1), self.at(i+1, j), self.at(i, j+1))
+
+
+class MatrixValueContainer():
+    '''Simple container for values in a MyMatrix object.
+
+    This class is intended to clean the syntax of setting
+    a new value in the MyMatrix object.
+
+    So we replace this:
+        matrix.get_value(i, j)
+        matrix.set_value(i, j, new_value)
+    with this:
+        matrix.at(i, j).value
+        matrix.at(i, j).set(new_value)
+
+    Args:
+        value : the value of the MatrixValue
+
+    '''
+    def __init__(self, value):
+        self.value = value
+
+    def set(self, new_value):
+        self.value = new_value
+
+
+class Point:
```

Comment:
Inside Util.py we already have a Vector2 class.  I think it would be cleaner to use that here rather than creating another one.